### PR TITLE
Refactor KeyValueDataSource interface

### PR DESCRIPTION
**Pull request by SergioDemianLerner**
Removal of execution code from the KeyValueDataSource interface, and move it to KeyValueDataSourceUtils. 
The refactor revealed a potential source of confusion where the key KEYVALUE_DATASOURCE_PROP_NAME was used for two unrelated things (database type determination and node configuration). 
Now there are two different keys: 
* PROPERTY_KEYVALUE_DATASOURCE  (for configuration)
* KEYVALUE_DATASOURCE_PROP_NAME (for database type determination)

## Description
We take out of the KeyValueDataSource interface the functionality to create a database  (LevelDB or RocksDB).

## Motivation and Context

- An interface should not contain execution code
- An abstract interface should not reference concrete implementations of that interface (LevelDB / RocksDB) because that defeats its abstraction purpose
- An identifier (KEYVALUE_DATASOURCE_PROP_NAME) should not be used for two unrelated things
- The KEYVALUE_DATASOURCE_PROP_NAME identifier should be private and not public, to enable changing the database type determination in the future.

## How Has This Been Tested?

All tests pass. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

### CODE REVIEW

File: rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSourceUtils.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+
+public class KeyValueDataSourceUtils {
+    static private String DB_KIND_PROPERTIES_FILE = "dbKind.properties";
```
Comment (Vovchyk) [inline rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSourceUtils.java:15]:
```suggestion
    private static final String DB_KIND_PROPERTIES_FILE = "dbKind.properties";
```

File: rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSourceUtils.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    static private String KEYVALUE_DATASOURCE_PROP_NAME = "keyvalue.datasource";
```
Comment (Vovchyk) [inline rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSourceUtils.java:16]:
```suggestion
    private static final String KEYVALUE_DATASOURCE_PROP_NAME = "keyvalue.datasource";
```

**Review by Vovchyk — CHANGES_REQUESTED** [review]
@SergioDemianLerner , thanks for the effort.

Could you please address Sonar complaints?

**Review by illuque-iov — APPROVED** [review]

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Review by Vovchyk — APPROVED** [review]

**Pull request merged**